packages/scenes/src/components/VizPanel/VizPanelRenderer.tsx

```diff
@@ -143,6 +144,11 @@ export function VizPanelRenderer({ model }: SceneComponentProps<VizPanel>) {
   const context = model.getPanelContext();
   const panelId = model.getLegacyPanelId();
 
+  let datasource: DataSourceRef | null | undefined = null
+  if (data.request?.targets?.length??0 > 0) {
+    datasource = data.request?.targets[0].datasource
```

Comment:
`datasource = data.request?.targets?.[0]?.datasource`
(instead of having the if condition at all)